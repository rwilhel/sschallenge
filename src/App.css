.App {
  text-align: center;
}

.App-header {
  background-color: #373737;
  height: 50px;
  padding: 20px;
  color: #C0B283;
  text-align: left;
}

.App-title {
  font-size: 1.5em;
  font-family: Verdana;
  display: inline-block;
  height: 100%;
  vertical-align: middle;
}

.App-intro {
  font-size: large;
}

.page {
  margin: 20px;
}

.interactions {
  text-align: center;
}

.table {
  margin: 20px 0;
}

.table-header {
  display: flex;
  line-height: 24px;
  font-size: 16px;
  padding: 0 10px;
  justify-content: space-between;
}

.table-empty {
  margin: 200px;
  text-align: center;
  font-size: 16px;
}

.table-row {
  display: flex;
  line-height: 30px;
  white-space: nowrap;
  padding: 10px;
  margin: 60px;
  background: #F4F4F4;
  border: 1px solid #DCD0C0;
  width: 90%;
}
