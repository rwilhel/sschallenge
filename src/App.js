import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      searchAddress: '',
    };

    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onSearchChange(event) {
    this.setState({ searchAddress: event.target.value });
  }

  render() {
    const { searchAddress, results } = this.state;
    return (
      <div className="App">
        <Search
          value={searchAddress}
          onChange={this.onSearchChange}
        />
        <Table
          results={results}
          pattern={searchAddress}
          onDismiss={this.onDismiss}
        />
      </div>
    );
  }
}

export default App;
