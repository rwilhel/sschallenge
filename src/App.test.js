import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App, { Search, Button, Table } from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Search', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Search>Search</Search>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Delete</Button>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Button>Delete</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
