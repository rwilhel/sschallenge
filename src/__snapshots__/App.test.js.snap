// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots 1`] = `null`;
